src/ui/buttons/buttons.jsx

```diff
@@ -150,6 +152,28 @@ export function validateButtonProps(props: ButtonPropsInputs) {
   normalizeButtonProps(props);
 }
 
+function calculateMessagePosition({ message, showPoweredBy, layout }): string {
+  if (!message) {
+    return "none";
```

Comment:
option to add a `NONE` constant here - https://github.com/paypal/paypal-checkout-components/blob/fbe73816fafc65d6fde81bb43d4107548e6effd7/src/constants/button.js#L77-L80 (for consistency) and change the return type to `$Keys<MESSAGE_POSITION>;`  (just kind of cool)

Reply:
We do not want `NONE` to be considered a valid input option which `MESSAGE_POSITION` is currently used for as a type. It's purely an internal computed value. Would it be beneficial to have a separate constant to capture the possible internal values?

Reply:
ah that's a good point! I think if it were me I would leave it as-is. one other idea just for fun would be to make the return type `string | void` and the early return would return nothing/undefined. But I think I like what you have here better than that

Reply:
@jshawl Nate and I tried a few different approaches being mindful of type returned here; the `null` return approach was our first attempted strategy, and if Flow would accommodate 'null', we would gladly go with that option.  But it didn't exactly mind our logic correctly: it tried to suggest `null` should be a possible value in the `Message` component where it cannot logically be passed in due to logic on [lines 302 and 359](https://github.com/paypal/paypal-checkout-components/pull/2335/files/fbe73816fafc65d6fde81bb43d4107548e6effd7#diff-3fba3feb650bae3f6bfc93d232bac1d8cfde682d5bfffbdd40d8cdfeda7fcd8aR302).  And it seemed wrong to abide that workaround by accommodating `null` inside Message where it could not appear.

We tried a few other typing workarounds, and finally settled on this being the least of the evils.

Reply:
this is flow's fault 😂